### PR TITLE
feat(web): Support InteractionStates like hovered and focused for web

### PR DESCRIPTION
**Pull request by theonetheycallneo**
Adding support for web interaction states (like hovered and focused):

https://necolas.github.io/react-native-web/docs/pressable/#examples

Resolves https://github.com/MarceloPrado/flash-calendar/issues/58

### CODE REVIEW

**Comment by theonetheycallneo** [comment]
This is based from https://github.com/MarceloPrado/flash-calendar/pull/62
Once that's merged, I'll clean this up as well.

**Comment by MarceloPrado** [comment]
could you rebase this PR?

**Comment by theonetheycallneo** [comment]
> could you rebase this PR?

@MarceloPrado rebased the latest `main`. Let me know what you think.

File: packages/flash-calendar/src/components/CalendarItemDay.tsx [hunk 1]
```diff
@@ -8,6 +8,14 @@ import type { CalendarDayMetadata } from "@/hooks/useCalendar";
...
 import { useTheme } from "@/hooks/useTheme";
 
+// react-native-web/overrides.ts
+declare module "react-native" {
+  interface PressableStateCallbackType {
+    hovered?: boolean;
+    focused?: boolean;
+  }
+}
+
```
Comment (theonetheycallneo) [inline packages/flash-calendar/src/components/CalendarItemDay.tsx:18]: @MarceloPrado Perhaps there's a better place to put custom types...

**Review by theonetheycallneo — COMMENTED** [review]
@MarceloPrado Left you 1 comment above^

**Comment by theonetheycallneo** [comment]
Closing in favor of this: https://github.com/MarceloPrado/flash-calendar/pull/68

**Pull request closed without merging**